core/extraction/towns_fund.py

```diff
@@ -547,6 +553,38 @@ def extract_funding_data(df_input: pd.DataFrame, project_lookup: dict) -> pd.Dat
 
     df_funding = convert_financial_halves(df_funding, "Reporting Period")
     df_funding.reset_index(drop=True, inplace=True)
+
+    # drop always unused cells TF funding before 2020
+    unused_mask = df_funding.loc[(df_funding["Secured"].isna()) & (df_funding["Start_Date"].isna())]
+    df_funding.drop(unused_mask.index, inplace=True)
+
+    if list(project_lookup.values())[0].split("-")[0] == "HS":
+        unused_fhsf_mask = df_funding.loc[
+            # drop unused FHSF Questions
+            (
+                df_funding["Funding Source Name"].isin(
+                    [
+                        "Town Deals 5% CDEL Pre-Payment",
+                        "Towns Fund RDEL Payment which is being utilised on TF project related activity",
+                        "How much of your RDEL forecast is contractually committed?",
+                    ]
+                )
+            )
+            |
+            # drop unused FHSF cells
+            (df_funding["Secured"].isna()) & (df_funding["Start_Date"] > end_of_reporting_period + timedelta(days=1))
+        ]
+        df_funding.drop(unused_fhsf_mask.index, inplace=True)
+
+    if list(project_lookup.values())[0].split("-")[0] == "TD" and check_programme_only:
+        unused_td_mask = df_funding.loc[
+            # drop unused TD Questions in the case of Programme Only
+            (df_funding["Funding Source Name"].isin(["Town Deals 5% CDEL Pre-Payment"]))
```

Comment:
The only problem with this approach is that users could theoretically legitimately add, for example, "Town Deals 5% CDEL Pre-Payment" as a "custom" funding source name, and in this case should be able to enter these fields without them being dropped? I guess we could just not worry about that very unlikely edge-case? Might be worth checking with a BA?

Reply:
did we just decide that we'd leave this as is?

Reply:
I've added some conditional logic to ensure: df_funding["Funding Source Type"] == "Towns Fund" which should ensure that "Other funding sources" are unaffected